src/utils/utils.ts

```diff
@@ -0,0 +1,6 @@
+export const isValidKey = <T extends object, Q extends keyof T>(
+  key: keyof T
+): key is Q => {
+  const runtimeKeys = Object.keys({} as T) as (keyof T)[];
```

Comment:
This won't work. On runtime every type will be stripped and this variable will be always empty array.